### PR TITLE
Drone plugin first implementation

### PR DESCRIPTION
**Pull request by athieriot**
Simple implementation using [PromiseSftp](https://www.npmjs.com/package/promise-sftp).

I'm using it via Password only at the moment so Private Key based authentication hasn't been tested properly.

### CODE REVIEW

File: DOCS.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Use this plugin to publish artifacts from the build to an SFTP server
+You can override the default configuration with the following parameters:
+
+* `host` - Server host
+* `port` - Server port, default to 22
+* `username` - Server username, default to blank
+* `password` - Password for password-based authentication
+* `privateKey` - String containing the private key for key-based authentication
+* `privateKeyFile` - Path to a private key (Used in conjonction with `volumes`) for key-based authentication
```
Comment (bradrydzewski) [inline DOCS.md]: good catch from @tboerger  ... do you mind if we stick to snake case for consistency with the other plugins?

File: DOCS.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+Use this plugin to publish artifacts from the build to an SFTP server
+You can override the default configuration with the following parameters:
+
+* `host` - Server host
+* `port` - Server port, default to 22
+* `username` - Server username, default to blank
+* `password` - Password for password-based authentication
+* `privateKey` - String containing the private key for key-based authentication
```
Comment (bradrydzewski) [inline DOCS.md]: also drone will automatically provide a public / private key pair for each repository (viewable in the repository settings screen in the drone UI). This is provided in `workspace.keys.private` ... for may of the other plugins (like rsync and ssh) we use this key by default.

for reference https://github.com/drone-plugins/drone-rsync/blob/master/main.go#L202

File: DOCS.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+Use this plugin to publish artifacts from the build to an SFTP server
+You can override the default configuration with the following parameters:
+
+* `host` - Server host
+* `port` - Server port, default to 22
+* `username` - Server username, default to blank
+* `password` - Password for password-based authentication
+* `destination_path` - Target path on the server, default to '/'
```
Comment (tboerger) [inline DOCS.md]: Maybe just path? No strong think, just some NITPICK

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    tag: latest
+    when:
+      branch: master
+
+plugin:
+  name: Sftp
+  desc: Publish files to an SFTP server
+  type: publish
+  image: plugin/drone-sftp
+  labels:
```
Comment (bradrydzewski) [inline .drone.yml]: note this will need to be `plugins/drone-sftp` (with an `s`)

**Comment by tboerger** [comment]
LGTM

**Comment by tboerger** [comment]
@nlf do you want to take a look at it as well?

File: DOCS.md (hunk 3 above)
Comment (athieriot) [inline DOCS.md]: Wanted to suggest it's the path on the remote server though.

**Comment by bradrydzewski** [comment]
LGTM

**Pull request merged**